### DISCUSSION TITLE
React-Admin Provider

### DISCUSSION THREAD

**Discussion by osseonews** [nhost/nhost#715] [Ideas]
If you want to increase adoption of your platform, I'd recommend building a dataprovider and authprovider using nHost for https://github.com/marmelab/react-admin. The Hasura dataprovider offered there is not great, and doesn't come with authentication or files. The supabase provider in react-admin provides everything. Your CRM example would be much more complete if you use React-Admin, as much of what you are building is already built with React-Admin.

**Comment by elitan** [comment]
That's a great idea. If you or anyone want to start implementing it feel free. Otherwise we'll prioritize it internally.

**Comment by osseonews** [comment]
@elitan React-Admin already has a Hasura data provider here: https://github.com/hasura/ra-data-hasura - so it should be super easy to integrate Nhost into this. The trouble we are having is setting the Authorization header in this Hasura provider with the bearer token from Nhost. Just not sure how to set it up to get it working properly. Maybe someone from your team can fork the ra-data-hasura and show a basic set up with Nhost authorization. Thanks.

**Reply by elitan** [reply]
Where do you specifically have issues? Any specific error message you're encountering?

It seems like you can add authentication headers here: https://github.com/hasura/ra-data-hasura#adding-authentication-headers

Either use the admin secret from your Nhost project. Or use an access token for a signed-in user. 

Something like this:

```js
  'Authorization': `Bearer ${nhost.auth.getAccessToken()}`,
```

**Reply by osseonews** [reply]
React-admin is a client-side application so using the admin secret is not really a secure option. As for the access token, we were using React-Admin's own Auth provider implementation (https://marmelab.com/react-admin/AuthProviderWriting.html) and we had trouble integrating it with the Nhost token that was generated and passing it to the dataprovider. I'm thinking it might just be easier to wrap the React-Admin is Nhost's own authentication.

**Reply by azlekov** [reply]
@osseonews after all did you have some success with the NHost authentication for React-Admin? Are you using ra-data-hasura or something else?

**Reply by osseonews** [reply]
No. I just use ra-data-hasura and then use the nhost credentials for hasura. I kind of view Nhost as just a hasura host and I don't really have success with their own libraries

**Reply by azlekov** [reply]
@osseonews are you using the admin secret or the NHost JS library and Bearer token?

**Reply by osseonews** [reply]
I try to never use the secret except on local development. So, I just create a Bearer token with admin role.
